sqlalchemy_hana/dialect.py

```diff
@@ -239,7 +239,8 @@ def _get_server_version_info(self, connection):
         pass
 
     def _get_default_schema_name(self, connection):
-        return self.normalize_name(connection.engine.url.username.upper())
+        # return self.normalize_name(connection.engine.url.username.upper())
+        return connection.execute("SELECT CURRENT_USER FROM DUMMY").fetchone()[0]
```

Comment:
Please use the CURRENT_SCHEMA function. Also I think it is necessary to use normalize_name.

Reply:
Fine with self.normalize_name.

But regarding the current schema vs current user, I have some interrogations.

I know that the `CURRENT_SCHEMA` doesn't reflect the default schema of a HANA user.
A user in HANA always have it's own default schema (named by the user id), and there is no way to change the user default schema.

However, during a session you can switch schema and avoid the schema prefixing using a set schema

And the [doc](http://docs.sqlalchemy.org/en/latest/core/reflection.html) says:
`default_schema_name`: Return the default schema name presented by the dialect for the current engine’s database user. E.g. this is typically public for PostgreSQL and dbo for SQL Server.